File: src/AgriApp.Application/Services/GstCalculator.cs
namespace AgriApp.Application.Services;

public class GstBreakdown
{
    public decimal BaseAmount { get; set; }
    public decimal Cgst { get; set; }
    public decimal Sgst { get; set; }
    public decimal TotalGst { get; set; }
    public decimal GrandTotal { get; set; }
}

public static class GstCalculator
{
    private const decimal CgstRate = 0.09m;
    private const decimal SgstRate = 0.09m;

    public static GstBreakdown Calculate(decimal baseAmount)
    {
        var cgst = Math.Round(baseAmount * CgstRate, 2, MidpointRounding.AwayFromZero);
        var sgst = Math.Round(baseAmount * SgstRate, 2, MidpointRounding.AwayFromZero);
        var totalGst = cgst + sgst;
        var grandTotal = baseAmount + totalGst;

        return new GstBreakdown
        {
            BaseAmount = baseAmount,
            Cgst = cgst,
            Sgst = sgst,
            TotalGst = totalGst,
            GrandTotal = grandTotal
        };
    }

    public static GstBreakdown CalculateRental(decimal hourlyRate, decimal hours)
    {
        var baseAmount = Math.Round(hourlyRate * hours, 2, MidpointRounding.AwayFromZero);
        return Calculate(baseAmount);
    }
}

File: artifacts/api-server/src/application/services/gstCalculator.ts
const GST_RATE = 0.18;
const CGST_RATE = 0.09;
const SGST_RATE = 0.09;

export interface GstBreakdown {
  baseAmount: number;
  cgst: number;
  sgst: number;
  totalGst: number;
  grandTotal: number;
}

export function calculateGst(baseAmount: number): GstBreakdown {
  const cgst = parseFloat((baseAmount * CGST_RATE).toFixed(2));
  const sgst = parseFloat((baseAmount * SGST_RATE).toFixed(2));
  const totalGst = parseFloat((cgst + sgst).toFixed(2));
  const grandTotal = parseFloat((baseAmount + totalGst).toFixed(2));

  return {
    baseAmount,
    cgst,
    sgst,
    totalGst,
    grandTotal,
  };
}

export function calculateRentalWithGst(hourlyRate: number, hours: number): GstBreakdown {
  const baseAmount = parseFloat((hourlyRate * hours).toFixed(2));
  return calculateGst(baseAmount);
}
